### PR TITLE
FACT-178 - Add court facilities tab to admin portal

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link ###

https://tools.hmcts.net/jira/browse/FACT-178

### Change description ###

Add court facilities tab to admin portal

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/app/controller/courts/CourtFacilitiesController.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    };
+
+    res.render('courts/tabs/facilitiesContent', pageData);
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+    let courtFacilities = req.body.courtFacilities as Facility[] ?? [];
+    courtFacilities.forEach(ot => {
+      ot.isNew = (ot.isNew === true) || ((ot.isNew as any) === 'true');
+      // Workaround for the issue where the the empty row is removed after the previous name selection.
```
Comment (Majid-justice) [inline src/main/app/controller/courts/CourtFacilitiesController.ts:63]: is it possible to give ot.isNew a type such as String and not any?

File: src/main/app/controller/courts/CourtFacilitiesController.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    await req.scope.cradle.api.updateCourtFacilities(req.params.slug, courtFacilities)
+      .then((value: Facility[]) => this.get(req, res, true, [], value))
+      .catch(() => this.get(req, res, false, [this.updateErrorMsg], courtFacilities));
+  }
+
+  public async addRow(req: AuthedRequest, res: Response): Promise<void> {
+    const courtFacilities = req.body.courtFacilities as Facility[] ?? [];
+    courtFacilities.forEach(ot => ot.isNew = (ot.isNew === true) || ((ot.isNew as any) === 'true'));
+    this.addEmptyFormsForNewEntries(courtFacilities);
```
Comment (Majid-justice) [inline src/main/app/controller/courts/CourtFacilitiesController.ts:97]: same here is it possible to give ot.isNew a type such as String and not any?

File: src/main/app/controller/courts/CourtFacilitiesController.ts (hunk 1 above)
Comment (KianKwa) [reply src/main/app/controller/courts/CourtFacilitiesController.ts:63]: I updated this one. Need to convert to unknown first then convert to string. If not typescript will throw an error and says can't convert from boolean to string directly. Not sure if this new way is better.

File: src/main/app/controller/courts/CourtFacilitiesController.ts (hunk 2 above)
Comment (KianKwa) [reply src/main/app/controller/courts/CourtFacilitiesController.ts:97]: Same as above

**Review by Majid-justice — APPROVED** [review]

**Pull request merged**